generate/templates/manual/revwalk/file_history_walk.cc

```diff
@@ -0,0 +1,270 @@
+
+  FileHistoryWalkBaton* baton = new FileHistoryWalkBaton;
+
+  baton->error_code = GIT_OK;
+  baton->error = NULL;
+  String::Utf8Value from_js_file_path(info[0]->ToString());
+  baton->file_path = strdup(*from_js_file_path);
+  baton->max_count = (unsigned int)info[1]->ToNumber()->Value();
+  baton->out = new std::vector< std::pair<git_commit *, std::pair<char *, git_delta_t> > *>;
+  baton->out->reserve(baton->max_count);
+  baton->walk = Nan::ObjectWrap::Unwrap<GitRevwalk>(info.This())->GetValue();
+
+  Nan::Callback *callback = new Nan::Callback(Local<Function>::Cast(info[2]));
+  FileHistoryWalkWorker *worker = new FileHistoryWalkWorker(baton, callback);
+  worker->SaveToPersistent("fileHistoryWalk", info.This());
+
+  Nan::AsyncQueueWorker(worker);
+  return;
+}
+
+void GitRevwalk::FileHistoryWalkWorker::Execute()
+{
+  git_repository *repo = git_revwalk_repository(baton->walk);
+  git_oid *nextOid = (git_oid *)malloc(sizeof(git_oid));
+  giterr_clear();
+  for (
+    unsigned int i = 0;
+    i < baton->max_count && (baton->error_code = git_revwalk_next(nextOid, baton->walk)) == GIT_OK;
+    ++i
+  ) {
+    // check if this commit has the file
+    git_commit *nextCommit;
+
+    if ((baton->error_code = git_commit_lookup(&nextCommit, repo, nextOid)) != GIT_OK) {
+      break;
+    }
+
+    git_tree *thisTree, *parentTree;
+    if ((baton->error_code = git_commit_tree(&thisTree, nextCommit)) != GIT_OK) {
+      git_commit_free(nextCommit);
+      break;
+    }
+
+    git_diff *diffs;
+    git_commit *parent;
+    unsigned int parents = git_commit_parentcount(nextCommit);
+    if (parents > 1) {
+      git_commit_free(nextCommit);
+      continue;
+    } else if (parents == 1) {
+      if ((baton->error_code = git_commit_parent(&parent, nextCommit, 0)) != GIT_OK) {
+        git_commit_free(nextCommit);
+        break;
+      }
+      if (
+        (baton->error_code = git_commit_tree(&parentTree, parent)) != GIT_OK ||
+        (baton->error_code = git_diff_tree_to_tree(&diffs, repo, parentTree, thisTree, NULL)) != GIT_OK
+      ) {
+        git_commit_free(nextCommit);
+        git_commit_free(parent);
+        break;
+      }
+    } else {
+      if ((baton->error_code = git_diff_tree_to_tree(&diffs, repo, NULL, thisTree, NULL)) != GIT_OK) {
+        git_commit_free(nextCommit);
+        break;
+      }
+    }
+
+    if ((baton->error_code = git_diff_find_similar(diffs, NULL)) != GIT_OK) {
+      git_commit_free(nextCommit);
+      git_commit_free(parent);
+      break;
+    }
+
+    bool flag = false;
+    bool doRenamedPass = false;
+    unsigned int numDeltas = git_diff_num_deltas(diffs);
+    for (unsigned int j = 0; j < numDeltas; ++j) {
+      git_patch *nextPatch;
+      baton->error_code = git_patch_from_diff(&nextPatch, diffs, j);
+
+      if (baton->error_code < GIT_OK) {
+        break;
+      }
+
+      if (nextPatch == NULL) {
+        continue;
+      }
+
+      const git_diff_delta *delta = git_patch_get_delta(nextPatch);
```

Comment:
Yea, thanks for linking this in case anyone didn't know. the delta is attached to the patch object. When we free the patch it is freed, as well.